src/plugins/content_management/public/content_client/content_client_mutation_hooks.tsx

```diff
@@ -18,3 +18,21 @@ export const useCreateContentMutation = <I extends CreateIn = CreateIn, O = unkn
     },
   });
 };
+
+export const useUpdateContentMutation = <I extends UpdateIn = UpdateIn, O = unknown>() => {
```

Comment:
I think we want to expose types that unique content can use to declare their mutation (and avoid the problem of declaring the generic everywhere).

Looking at the `useCreateContentMutation` we would expose

```ts
export type CreateContentMutationHook<
  I extends CreateIn = CreateIn,
  O = unknown
> = UseMutationResult<O, unknown, I, unknown>;
```

Each content can then expose their mutation types

```ts
type CreateFooMutation = CreateContentMutationHook<
  {
    contentTypeId: 'foo';
    data: { title: string }; // Shape of the input
  },
  { result: 'success' | 'error' }
>;
```

And when consumed it can be used like this

```ts
const createFoo: CreateFooMutation = useCreateContentMutation();
```

It is properly typed

Reply:
If that works we can apply the same for queries

Reply:
Sounds good. I'll try and see if there are any issues